.github/workflows/release_ci.yml

```diff
@@ -97,6 +98,7 @@ jobs:
         Get-ChildItem ./src/OrchardCore.Cms.Web/App_Data -Recurse | Remove-Item
         Get-ChildItem ./src/OrchardCore.Cms.Web/App_Data_Tests -Recurse | Remove-Item
         dotnet publish -c Release -o ./.build/release --no-build --no-restore --framework net6.0
+        Copy-Item -Path "./src/OrchardCore.Cms.Web/Localization" -Destination "./.build/release/Localization" -Recurse
```

Comment:
Sound like an issue with our dotnet publish that it's not copied automatically.

Reply:
It's more because of the previous `dotnet build` that doesn't copy them automatically because here we just take the files from the previous build made in the "build step". Here, we save time by doing `dotnet publish --no-build` which will take the files previously built in the /src/* folder.

Reply:
So I just tried locally a `dotnet build --configuration Release --framework net6.0` directly on the root folder of my solution just after doing a `git clean -xdf` and the Localization folder gets updated with .po files at the end of the build process. So, I'm questioning myself now if this folder is there after doing the build on the CI. We could do a `cd ./src/OrchardCore.Cms.Web/Localization/fr` for example.

Reply:
But I think that this folder is not taken into consideration because on the dotnet publish we say --no-restore and --no-build
So it takes only the files that would be generated without building and restoring which is what is required to get the Localization folder. It kind of does its job correctly.

Reply:
I need to take a look at #4088

Reply:
![image](https://user-images.githubusercontent.com/3228637/143300579-7be06655-e9bf-485e-b6d5-ac32ec4feee1.png)

You can see here that the `wwwroot` folder is there but it may be created at the same time than creating the ìs-cache` folder.

Reply:
Here locally on my PC when I execute

`dotnet publish -c Release -o ./.build/release --no-build --no-restore --framework net6.0`

It doesn't create any `Localization` folder but does create a `wwwroot` folder.

![image](https://user-images.githubusercontent.com/3228637/143301342-34d8ec91-648c-4be2-a406-4a882ca10c55.png)

Reply:
Another attempt here : 

![image](https://user-images.githubusercontent.com/3228637/143302431-e519d356-e299-4872-84bf-df7c45b7b10e.png)

Removing Localization in the first ItemGroup items. Redo a dotnet publish and still nothing.

Reply:
Can't now but I will take a look this night

We have the following msbuild target, maybe the problem is that it only executes after a buid


    <Target
      Name="CopyPackageTranslationFiles"
      Condition="'$(IsCrossTargetingBuild)' != 'true' and ('$(IsCrossTargetedBuild)' != 'true' or '$(TargetFrameworks' == '' or '$(TargetFrameworks.IndexOf($(TargetFramework)))' == '0')"
      AfterTargets="Build"> <== Maybe the problem

      <Message Text="Copying translation files: $(MSBuildProjectName)" Importance="high" />
      <Copy
        SourceFiles="%(PackageTranslationFiles.FullPath)"
        DestinationFolder="Localization\%(RecursiveDir)"
        Condition="'@(PackageTranslationFiles)' != ''"
        SkipUnchangedFiles="true" />

      <ItemGroup>
        <LocalizationFiles Include="Localization\**" Exclude="$(DefaultItemExcludes)" />
        <ResolvedFileToPublish Include="@(LocalizationFiles)">
          <RelativePath>Localization\%(RecursiveDir)%(Filename)%(Extension)</RelativePath>
        </ResolvedFileToPublish>
     </ItemGroup>
    </Target>

Reply:
Ok, so the way it works is that here we remove any App_Data and Localization folder in every project that imports it.
This is kind of a reset function for all projects.

https://github.com/OrchardCMS/OrchardCore/blob/aa724a3a56c71f4d32b1973e4a2e73255b90d5cf/src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.targets#L3-L8

But then in the OrchardCore.Cms.Web.csproj we override these settings here : 

https://github.com/OrchardCMS/OrchardCore/blob/aa724a3a56c71f4d32b1973e4a2e73255b90d5cf/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj#L6-L19

And you can see that we import the targets file before the override in the specific project.

We do this also in the OrchardCore.Application.Pages test project.

I'm inspecting the resulting OrchardCore.Cms.Web.exe and .dll with dotPeek and there are no resource files added to the assembly. Which means it's ok.

Reply:
So the wwwroot folder and Localizations folder are really only added when doing a dotnet publish only. At least this part still works.

Reply:
```
 <ItemGroup> 
   <Compile Remove="App_Data\**;Localization\**" /> This removes from compilation
   <Content Remove="App_Data\**;Localization\**" /> This removes from publish
   <EmbeddedResource Remove="App_Data\**;Localization\**" /> This removes from assembly as embedded resource
   <None Remove="App_Data\**;Localization\**" /> This removes from assembly as embedded resource
 </ItemGroup> 
```

Reply:
I don't know, this is SO weird! I give up for now. 🙈

Reply:
@jtkech just saw your message. Yes, but here when we do a dotnet publish --no-build it won't recompile anything. Just use what has already been compiled in the /src folder. So the msbuild target here is probably not used at all when doing the dotnet publish.

Reply:
Yes this is what I meant

Reply:
Here is another test.

```
git clean -xdf
dotnet build --configuration Release --framework net5.0
dotnet publish -c Release -o ./.build/release --no-build --no-restore --framework net5.0
```

![image](https://user-images.githubusercontent.com/3228637/143310172-2648a8bc-0ccb-4ba0-bbd5-89e14ea8171a.png)

![image](https://user-images.githubusercontent.com/3228637/143310189-f4ec8a7b-422b-4eea-b0f4-a3f757900f73.png)

Same result. The Localization folder is not there. I will make an other test by moving to an older commit on the main branch before we merged the .NET 6.0 PR. and report here what I find.

Reply:
![image](https://user-images.githubusercontent.com/3228637/143310827-0b964a90-44db-4a98-870c-45dd78df0c31.png)

Same process and here the result : 

![image](https://user-images.githubusercontent.com/3228637/143310872-b99d56c0-1995-4e0e-80d6-32f97f16c901.png)

No Localization folder. Now I'm moving on testing older Docker images.

Reply:
RC2 image works 👍🏼 

![image](https://user-images.githubusercontent.com/3228637/143311401-bf38d55d-6b52-4e86-a5b2-58560edad40f.png)

1.1.0 doesn't work 👎🏼 

![image](https://user-images.githubusercontent.com/3228637/143311711-94310704-bec6-4c7b-a475-9294260c14c4.png)

Reply:
So the only thing I'm thinking right now is that `dotnet publish --no-build` used to trigger the .targets file that was downloading the Localization folder. And now with .NET 6.0 it doesn't anymore. So, that's an MSBuild issue, or actually, it has been fixed because --no-build makes sense to not trigger any building task.

Reply:
My fix doesn't hurt to be added to make a second pass and to be sure this Localization folder is there.

Reply:
Ok, good job. But then we should have a hook to copy the localization files on publish too, when --no-build is used.

Reply:
I'm testing more and I found out that RC2 was using AppVeyor.